discovery/discovery.go

```diff
@@ -221,7 +221,7 @@ func (d *discovery) checkCluster() ([]*client.Node, int, uint64, error) {
 		if err == client.ErrInvalidJSON {
 			return nil, 0, 0, ErrBadDiscoveryEndpoint
 		}
-		if err == context.DeadlineExceeded {
+		if _, ok := err.(*client.ClusterError); ok {
```

Comment:
> I still think that it needs to print out detailed error to help user understand why it failed.

Printing out the error is one thing; exiting is another thing. We can print the error out for sure.